src/am_map.c

```diff
@@ -1618,23 +1618,64 @@ static void AM_drawWalls(void)
     {
       if ((lines[i].flags & ML_DONTDRAW) && !ddt_cheating)
         continue;
-      if (!lines[i].backsector)
       {
-        if //jff 4/23/98 add exit lines to automap
+        /* cph - show keyed doors and lines */
+        int amd;
+        if ((mapcolor_bdor || mapcolor_ydor || mapcolor_rdor) &&
+            !(lines[i].flags & ML_SECRET) &&    /* non-secret */
+            (amd = AM_DoorColor(lines[i].special)) != -1
```

Comment:
Is this code from PrBoom? Maybe we should make it more readable.

Reply:
I just suggest `int amd = AM_DoorColor(lines[i].special);` instead of `(amd = AM_DoorColor(lines[i].special)) != -1`. 

I tried `clang-format` a while ago, it only works well for the built-in styles (LLVM, Google, etc.) No matter how much I changed the [options](https://clang.llvm.org/docs/ClangFormatStyleOptions.html), it didn't work for MBF or Chocolate Doom style. However, I think we should reformat all new files and functions in the Chocolate Doom style.

Reply:
> I just suggest `int amd = AM_DoorColor(lines[i].special);` instead of `(amd = AM_DoorColor(lines[i].special)) != -1`.

Yes, that's better.

> I tried `clang-format` a while ago, [...], it didn't work for MBF or Chocolate Doom style. However, I think we should reformat all new files and functions in the Chocolate Doom style.

When using Chocolate's `.clang-format` file, I found the results quite usable. For example, I formatted the `src/p_extnodes.c` file this way when I rewrote it for XGLN nodes support.

Reply:
These are all new files, quite a lot:

```
$ git diff --diff-filter=A --name-only upstream:Source master:src
CMakeLists.txt
beta.h
cross.h
d_iwad.c
d_iwad.h
d_loop.c
d_loop.h
d_quit.c
d_quit.h
dogs.h
doomkeys.h
dsdhacked.c
dsdhacked.h
font.h
hu_obituary.c
hu_obituary.h
i_3dsound.c
i_endoom.c
i_endoom.h
i_flmusic.c
i_glob.c
i_glob.h
i_input.c
i_input.h
i_macmusic.c
i_mbfsound.c
i_oalmusic.c
i_oalmusic.h
i_oalsound.c
i_oalsound.h
i_oalstream.h
i_oplmusic.c
i_pcsound.c
i_printf.c
i_printf.h
i_sndfile.c
i_sndfile.h
i_timer.c
i_timer.h
i_winmusic.c
i_xmp.c
icon.c
m_input.c
m_input.h
m_io.c
m_io.h
m_misc2.c
m_misc2.h
m_snapshot.c
m_snapshot.h
manifest.xml
memio.c
memio.h
midifallback.c
midifallback.h
midifile.c
midifile.h
mus2mid.c
mus2mid.h
net_client.c
net_client.h
net_common.c
net_common.h
net_dedicated.c
net_dedicated.h
net_defs.h
net_gui.c
net_gui.h
net_io.c
net_io.h
net_loop.c
net_loop.h
net_packet.c
net_packet.h
net_petname.c
net_petname.h
net_query.c
net_query.h
net_sdl.c
net_sdl.h
net_server.c
net_server.h
net_structrw.c
net_structrw.h
p_action.h
p_extnodes.c
p_extnodes.h
params.h
r_bmaps.c
r_bmaps.h
r_swirl.c
r_swirl.h
r_voxel.c
r_voxel.h
resource.rc.in
s_musinfo.c
s_musinfo.h
setup-res.rc.in
statdump.c
statdump.h
thermo.h
u_mapinfo.c
u_mapinfo.h
u_scanner.c
u_scanner.h
v_trans.c
v_trans.h
```

Reply:
> These are all new files, quite a lot:

Most of them are copied from Chocolate. Which makes inсonsistent files more annoying. So, I can reformat all the files I have created. I think we should keep r_voxel.c as is.

Reply:
There are some files that should be excluded from this list, e.g. `CMakeLists.txt`, `*.in`, `params.h`, `icon.c`. Not sure about `r_voxel.c`, though, where do we start with exceptions, where do we end?

Reply:
> Not sure about r_voxel.c, though, where do we start with exceptions, where do we end?

I think that's it, no more exceptions. r_voxel.c is a non-trivial file and very carefully formatted, it is clearly the author's intention.

Reply:
Alright, let's got for it then. Please do also add the config file for the formatting tool that you chose to use.